WordPressAuthenticator/Authenticator/WordPressAuthenticatorDelegateProtocol.swift

```diff
@@ -48,7 +48,7 @@ public protocol WordPressAuthenticatorDelegate: class {
     ///     - site: passes in the site information to the delegate method.
     ///     - onCompletion: Closure to be executed on completion.
     ///
-    func shouldPresentUsernamePasswordController(for siteInfo: WordPressComSiteInfo?, onCompletion: @escaping (Error?, Bool) -> Void)
+    func shouldPresentUsernamePasswordController(for siteInfo: WordPressComSiteInfo?, onCompletion: @escaping (Error?, Bool, UIViewController?) -> Void)
```

Comment:
This would break WPAuhenticator's public API. 

Also, the way I understand it, these three parameters are mutually exclusive. So it would probably be better if the signature of this method was actually something like:

`func shouldPresentUsernamePasswordController(for siteInfo: WordPressComSiteInfo?, onCompletion: @escaping (WPAuthenticatorResult) -> Void)`

Where:
```
enum WPAuthenticatorResult {
    case error(value: Error)
    case presentPasswordController(value: Bool)
    case injectViewController(value: UIViewController)
}
```

or similar.

Reply:
In WPiOS, `shouldPresentUsernamePasswordController` just calls `onCompletion`. 

```
func shouldPresentUsernamePasswordController(for siteInfo: WordPressComSiteInfo?, onCompletion: @escaping (Error?, Bool) -> Void) {
    onCompletion(nil, true)
}
```

So fixing it should be straightforward.

> So it would probably be better if the signature of this method was actually something like

That sounds reasonable.